game/block_game_agents_example.py

```diff
@@ -0,0 +1,103 @@
+from block_game import ACTIONS, P1, P2, BlockGameState, BlockGameMDP
+from block_game_agent import FixedPolicyBlockGameAgent, DynamicPolicyBlockGameAgent
+from block_game_tree import BlockGameTree, AVAILABLE
+from copy import deepcopy
+import numpy as np
+from simple_rl.run_experiments import play_markov_game
+from typing import List, Tuple
+
+block_game_tree = BlockGameTree()
+
+
+# -------------------------------------------------------------------
+# ------------------------ STATIC AGENTS ----------------------------
+# -------------------------------------------------------------------
+def minimax_func(reward_action_pairs: List[Tuple[Tuple[float, float], int]], turn: int) -> Tuple[Tuple[float, float], int]:
+    other_player = P1 if turn == P2 else P2
+    return min(reward_action_pairs, key=lambda x: x[0][other_player])
+
+
+minimax_agent = FixedPolicyBlockGameAgent(
+    minimax_func, 'Minimax', deepcopy(block_game_tree))
```

Comment:
Is the deepcopy needed? The agents shouldn't alter the game tree, just store their own desired actions for the game. Oh, I see you are storing ideal_reward in the tree node. Maybe store that in the agent instead so deepcopy is not needed?

Reply:
Alright, 1 sec

Reply:
Done